pennylane/devices/default_clifford.py

```diff
@@ -460,57 +519,239 @@ def simulate(
 
-                # Computing statevector via tableaus
-                elif isinstance(meas, StateMP):
-                    res = self._measure_state(tableau_simulator, circuit, global_phase)
+            # Computing observable expectation value using above classical shadows
+            elif isinstance(meas, ShadowExpvalMP):
+                results.append(self._measure_expval_shadow(stim_circuit, circuit, meas, stim))
 
-                # Computing purity via tableaus
-                elif isinstance(meas, PurityMP):
-                    res = self._measure_purity(meas, circuit)
+            elif isinstance(meas, ExpectationMP):
+                results.append(
+                    self._measure_expectation_samples(stim_circuit, meas_op, num_shots, sample_seed)
+                )
 
-                # Computing expectation values via measurement
-                elif isinstance(meas, ExpectationMP):
-                    res = self._measure_expectation(tableau_simulator, meas, stim)
+            elif isinstance(meas, VarianceMP):
+                meas_obs = qml.operation.convert_to_opmath(meas_op)
+                with qml.operation.use_new_opmath():
+                    meas_obs1 = meas_obs.simplify()
+                    meas_obs2 = (meas_obs1**2).simplify()
 
-                # Computing more measurements
-                else:
-                    raise NotImplementedError(
-                        f"default.clifford doesn't support the {type(meas)} measurement at the moment."
+                # use the naive formula for variance, i.e., Var(Q) = ⟨𝑄^2⟩−⟨𝑄⟩^2
+                vars = (
+                    self._measure_expectation_samples(
+                        stim_circuit, meas_obs2, num_shots, sample_seed
+                    )
+                    - self._measure_expectation_samples(
+                        stim_circuit, meas_obs1, num_shots, sample_seed
                     )
+                    ** 2
+                )
+                results.append(vars)
+
+            # Computing rest of the measurement by processing samples
+            else:
+                samples = qml.math.array(
+                    [
+                        self._measure_observable_sample(
+                            meas_op, stim_circuit, num_shots, sample_seed
+                        )[0]
+                    ]
+                )
+                wire_order = {wire: idx for idx, wire in enumerate(meas.wires)}
+                results.append(meas.process_samples(samples=samples, wire_order=wire_order))
+        return results
+
+    def measure_analytical(self, circuit, stim_circuit, tableau_simulator, global_phase, stim):
+        """Given a circuit, compute tableau and return the analytical measurement results."""
+        results = []
+        for meas in circuit.measurements:
+            # Computing density matrix via tableaus
+            if isinstance(meas, DensityMatrixMP):  # do first because it is a child of StateMP
```

Comment:
What would think of moving this logic to either a:

* single dispatch function with signature `(measurement, circuit, stim_circuit, tableau_simulator, global_phase)`

or 

* dictionary mapping type to callable?
```
measurement_map = {
    DensityMatrixMP: self._measure_density_matrix,
    StateMP: self._measure_state,
    ExpectationMP: self._measure_expecation,
    ...
}
results = []
for m in circuit.measurements:
     r = measurement_map[type(m)](measurement, circuit, stim_circuit, tableau_simulator, global_phase)
     results.append(r)
return tuple(results)
```

The pattern of extended `elif`'s can usually be made a lot more readable and maintainable by adhering to a fixed call signature and making the functions interchangeable.  Otherwise, it can be difficult to remember which function had which call signature.

Reply:
I did think about it but found it hard to come up with a single dispatch signature, so I had to resort to use the `elif`'s structure. However, I finally added dictionary mapping for stochastic and analytic measurements, as you suggested.